### PR TITLE
graph algo added

### LINKED ISSUE #674: [Feature Request]: Graphs ALgos and theorms in DSA
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Feature Description

adding graph algorithms like

1. Tarjan's Algorithm
2. prim's algo
3. multidtage graph algo

### Use Case

dsa graph docs

### Benefits

_No response_

### Add ScreenShots

_No response_

### Priority

High

### Record

- [X] I have read the Contributing Guidelines
- [X] I'm a GSSOC'24 contributor
- [X] I have starred the repository

### PR DESCRIPTION
**Pull request by Yashgabani845**
## Related Issue

Fixes #674 

## Description

added these 3 algo

1. tarjan
2. prims
3. multistage graph 

## Type of PR

- [x] Feature enhancement
- [x] Documentation update

## Screenshots / Videos (if applicable)
Screenshot (553)
Screenshot (554)
Screenshot (555)
Screenshot (557)
Screenshot (558)
Screenshot (559)
Screenshot (560)
Screenshot (561)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Here are some resources that may be helpful as you contribute to CodeHarborHub:
- [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [Markdown Guide](https://www.markdownguide.org/)
- [MDX Documentation](https://mdxjs.com/docs/)
- [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- Tabs: [Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex)

### CODE REVIEW

**Comment by Yashgabani845** [comment]
@Ajay-Dhangar 
added 3 graph algorithm with deep explanation and implementation in 4 languages

File: dsa/Algorithms/multistage-graph.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+2. Calculate the shortest path cost
+
+ for each vertex considering all possible paths.
+3. The shortest path cost from the source vertex will be the result.
+
+The shortest path cost is 17.
+
+## Conclusion
+
+The Multistage Graph Algorithm efficiently finds the shortest path in a graph divided into stages. It operates in O(V^2) time complexity, making it suitable for graphs with a structured stage-based layout. Understanding and implementing this algorithm is essential for solving complex optimization problems in various applications.
```
Comment (Ajay-Dhangar) [inline dsa/Algorithms/multistage-graph.md:225]: replace `O(V^2)` to `$O(V^2)$`

File: dsa/Algorithms/prims.md [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+1. Start with vertex 'A'.
+2. The edge with the smallest weight from 'A' is ('A', 'B', 1), add it to the MST.
+3. The next smallest edge is ('B', 'D', 2), add it to the MST.
+4. Finally, add edge ('B', 'C', 3) to the MST.
+
+The MST contains the edges: [('A', 'B', 1), ('B', 'D', 2), ('B', 'C', 3)].
+
+## Conclusion
+
+Prim's Algorithm is an efficient method for finding the Minimum Spanning Tree of a weighted undirected graph. It operates in O(E log V) time complexity using a priority queue, making it suitable for large graphs. Understanding and implementing this algorithm is fundamental for solving various network design problems and optimizing connectivity.
```
Comment (Ajay-Dhangar) [inline dsa/Algorithms/prims.md:324]: replace `O(E log V)` to `$O(E log V)$`

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by Ajay-Dhangar** [comment]
@Yashgabani845, could you please update it? Otherwise, I may need to close it.

**Comment by Yashgabani845** [comment]
@Ajay-Dhangar sorry sir  I am doing it rn

**Review by Yashgabani845 — COMMENTED** [review]
done sir

**Review by Ajay-Dhangar — APPROVED** [review]
Done

**Pull request merged**